content/reference/rest-api/engine-diagrams/versions.md

```diff
@@ -0,0 +1,45 @@
+---
+title: 'Engine Diagrams: versions'
+weight: 10
+
+menu:
+  main:
+    name: 'versions'
+    identifier: 'rest-api-engine-diagrams-versions'
+    parent: 'rest-api-engine-diagrams'
+    pre: 'Lists all available versions of an engine diagram.'
+---
+
+Lists all available versions of an engine project.
+
+# Request
+
+## HTTP Request
+
+```
+GET https://cawemo.com/api/v1/engine-projects/:project-name/engine-diagrams/:diagram-name/versions
```

Comment:
I'm wondering how we can relate 
- the :project-name to the engine name (see 'name' attribute of the '&lt;process-engine&gt;' tag https://docs.camunda.org/manual/7.15/reference/deployment-descriptors/tags/process-engine/#syntax-reference)
- the :diagram-name to the process definition key

Reply:
Strictly speaking in terms of the rest API and maybe with the risk of missing some context here, I would like to ask why would we use `:diagram_name` since this should be a unique identifier?

To explain better where my question is coming from: Cawemo today does not enforce unique project/folder/file names in an organization/project or folder/project or folder.

Reply:
In today's meeting we agreed on:

- using the project's id instead of the project's name
- renaming the diagram-name path variable to process-definition-key

> I would like to ask why would we use :diagram_name since this should be a unique identifier.

We clarified this question today - the motivation behind using a name (or the process definition key) instead of Cawemo's unique identifier is stemming from reproducing the behavior of the Camunda Platform engine where the process definition keys are treated as unique identifier. Cawemo's backend will make sure that there process definition key's are unique within the project scope.